### ISSUE TITLE
continue in case has undefined behaviour

### ISSUE THREAD

**Issue by TheLastProject** [koalaman/shellcheck#1576]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

printf "This\nis\na\ntest" | while IFS= read -r line; do
  echo "$line"

  case "$line" in
    *is*)
      # Matches is and This
      ;;
    *)
      # This is invalid
      continue
      ;;
  esac

  echo "$line"
done
```

```sh
$ bash test.sh
This
This
is
is
a
```
(Also happens with ksh and zsh on my machine)

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Continue in a case statement has undefined behaviour.

**Comment by koalaman** [comment]
This does appear to be correct and well defined according to 
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#break : the continue is enclosed in the `while` loop according to the definition. All the shells I tried behaved accordingly.

If you don't enclose it in a `while` loop the result is undefined, and in that case you already get a SC2105. 

Please reopen if I've misunderstood your point.
